.github/scripts/generate_binary_build_matrix.py

```diff
@@ -57,58 +77,82 @@
 ]
 
 
-def is_pull_request():
+def is_pull_request() -> bool:
     return False
     # return os.environ.get("GITHUB_HEAD_REF")
 
-def generate_matrix(include_cuda=True, include_rocm=False):
-    python_versions = FULL_PYTHON_VERSIONS
-    cuda_versions = CUDA_ARCHES
-    rocm_versions = ROCM_ARCHES
+
+def snip_if(is_pr: bool, l: List[str]) -> List[str]:
+    return [l[-1]] if is_pr else l
```

Comment:
This looks a bit confusing, I think you should change the `l` parameter to be something a bit more descriptive.

Perhaps a docstring would work as well?